lib/asn1/src/asn1_db.erl

```diff
@@ -82,11 +82,11 @@ loop(#state{parent = Parent, monitor = MRef, table = Table,
             includes = Includes} = State) ->
     receive
         {set, Mod, K2, V} ->
-            [{_, Modtab}] = ets:lookup(Table, Mod),
+            Modtab = ets:lookup_element(Table, Mod, 2),
```

Comment:
This is not fully equivalent. The old code will badmatch if the found element is a tuple of size >2, but the new code will pass.

Reply:
@Maria-12648430 The lookup is intended to always succeed. Testing the tuple size is only done because it was the most convenient way to get the second element. It is also an (accidental) assertion to crash if there is some strange bugs that stores garbage in the table. In practice, that assertion is very unlikely to ever fire because the table is only writeable by the process owning it.

@RobinMorisset On the other hand, this module is part of the ASN.1 compiler, so the optimization will give very little.

Reply:
I found this code by a naive grep for ets:lookup, I am quite willing to drop this change.
The change in this PR that I consider important is the application_controller:get_env one, as that function shows surprisingly heavily in some profiling I did.

Reply:
> I found this code by a naive grep for ets:lookup, I am quite willing to drop this change.

In that case, I suggest that you drop the changes in `asn1db`.